File: src/main/Java/boj_step20/boj25192.java
import java.io.*;
import java.util.HashSet;

public class boj25192 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = Integer.parseInt(reader.readLine());
        HashSet<String> chatLog = new HashSet<>();
        int count = 0;
        while (n-- > 0) {
            String chat = reader.readLine();
            if (chat.equals("ENTER")) chatLog.clear();
            else if (!chatLog.contains(chat)) {
                chatLog.add(chat);
                count++;
            }
        }
        writer.append(String.valueOf(count)).close();
    }
}

File: src/main/boj_step20/boj25192.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val n = r.readLine().toInt()
    val chatlog = HashSet<String>()
    var count = 0
    repeat(n) {
        val chat = r.readLine()
        if (chat == "ENTER") chatlog.clear()
        else if (chat !in chatlog) {
            count++
            chatlog.add(chat)
        }
    }
    w.write("${count}")
    w.close()
}